docs/current/sdk/elixir/756758-get-started.md

```diff
@@ -0,0 +1,87 @@
+---
+slug: /sdk/elixir/756758/get-started
+---
+
+# Get Started with the Dagger Elixir SDK
+
+## Introduction
+
+This tutorial teaches you the basic of using Dagger in Elixir. You will learn how to:
+
+- Install Elixir SDK
+- Create an Elixir CI task to test an application
+- Improve the Elixir CI tool to test the application against multiple Elixir and OTP versions
+
+## Requirements
+
+This tutorial assumes that:
+
+- You have a basic understanding of the Elixir programming language. If not, [read the Elixir learning](https://elixir-lang.org/learning.html).
+- You have an Elixir development environment with Elixir 1.14 or later and Erlang/OTP 25 or later. If not, install [Elixir](https://elixir-lang.org/install.html) and [Erlang/OTP](https://www.erlang.org/downloads).
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+
+## Step 1: Create an Elixir project
+
+Create an Elixir project with `mix`:
+
+```shell
+mix new a_project
+```
+
+## Step 2: Install the Dagger Elixir SDK
+
+Open `mix.exs` and add `{:dagger, "~> 0.8"}` to the list in `deps` function:
+
+```elixir
+def deps do
+  [
+    {:dagger, "~> 0.8", only: [:dev, :test]}
+  ]
+end
+```
+
+Then run `mix deps.get` to fetch the Elixir SDK from Hex.pm.
+
+## Step 3: Create the Mix task
+
+Create Elixir module at `lib/mix/tasks/a_project.test.ex`:
+
+```elixir file=snippets/get-started/step3/a_project.test.ex
+```
+
+This module performs the following operations:
+
+- It starts all applications that related to Dagger Elixir SDK with `Application.ensure_all_started(:dagger)`.
+- It creates a Dagger client with `Dagger.connect!/1`.
+- It uses the client's `Dagger.Client.host/1` and `Dagger.Host.directory/3` to address the host directory. Uses `exclude` in `Dagger.Host.directory/3` to filter unwanted files and directories.
+- It uses the client's `Dagger.Client.container/2` and `Dagger.Container.from/2` to initialize a new container and uses `hexpm/elixir:1.15.4-erlang-25.3.2.5-ubuntu-bionic-20230126` as a base image.
+- It uses `Dagger.Container.with_mounted_directory/3` to mount source files into container.
+- It uses `Dagger.Container.with_exec/3` to execute a command.
+- It uses `Dagger.Sync.sync/1` to force executing commands.
+- It uses `Dagger.close/1` to teardown the client.
+
+Run the Mix task by executing the command below from the project directory:
+
+```shell
+mix a_project.test
+```
```

Comment:
What are your thoughts about adding `dagger run` in front of this command @wingyplus? The output will look a lot better! FWIW, it will require downloading & installing the `dagger` CLI: https://docs.dagger.io/cli/465058/install

This output will become especially important when testing against multiple versions below.

Reply:
@wingyplus in my copy edit I have modified these commands to use `dagger run`, as this is now the recommended approach. Please double-check that the output is as expected.

Reply:
I update the output here but please note that we have https://github.com/dagger/dagger/issues/5666, it stuck when running with `dagger run` command.

Reply:
Works on both macOS & NixOS for me: https://github.com/dagger/dagger/issues/5666#issuecomment-1740760580